alc/backends/pulseaudio.cpp

```diff
@@ -927,9 +934,13 @@ bool PulsePlayback::reset()
         flags &= ~PA_STREAM_EARLY_REQUESTS;
         flags |= PA_STREAM_ADJUST_LATENCY;
     }
+    if(!mDevice->Flags.get<SampleTypeRequest>() && !mSettingFormatFailed)
+        flags |= PA_STREAM_FIX_FORMAT;
     if(GetConfigValueBool(mDevice->DeviceName.c_str(), "pulse", "fix-rate", 0) ||
        !mDevice->Flags.get<FrequencyRequest>())
         flags |= PA_STREAM_FIX_RATE;
+    if(!mDevice->Flags.get<ChannelsRequest>() && !mSettingChannelsFailed)
+        flags |= PA_STREAM_FIX_CHANNELS;
```

Comment:
This will cause the mixer to write incorrect channels (different order, too many, or too few) if it ends up getting changed by the server. Similar for the sample format, writing samples that are too large, too small, different signedness, etc, if it gets changed.

Reply:
How channels change should be handled? Should some buffer attributes be changed?